### ISSUE TITLE
test data后半段不理想优化

### ISSUE THREAD

**Issue by zhuanxuhit** [zhuanxuhit/dlnd-your-first-neural-network#2]
If you look at the results, you can clearly find your model is doing better in the first ten days than the last ten days. Remember the way you divided the training data set and test data set. You leave the data in the last 21 date as test data and train the model use the data in the earlier time period. So it makes sense that your model are doing good in the date that closer to the date in your training data and worse when it goes farther beyond your training data period. We can also argue the end of the Dec is Chrismas holiday and one can expect the customer behavior will get different than usual days. You can see it as a kind of over fitting. To improve the model, you can randomly split the data into training and testing data set and ignore the time ordering and rebuild your model.

There is no direct rule to do it, but hyper parameter optimization is one of the methods.
Play with the number of hidden layers, learning rate, the activation function etc...

Try checking the accuracy and improving the results using the following grid,
Learning rate,: 0.001,0.005, 0.1, 0.2
Number of hidden layers: 10,20,40,80,160
Activation function ,: sigmoid, tanh

Try all the above combinations and see which gives the best results..
